tools/gui/qtkontakt.py

```diff
@@ -407,6 +429,47 @@ def __get_impfzentren_plz(self, plzList : list) -> str:
             plz_zentrum_raw += plz + ', '
         return plz_zentrum_raw[:-2]
 
+    def __get_vermittlungscodes(self) -> list:
+        """
+        Erstellt ein Liste mit Vermittlungscodes aus der GUI.
+        Unvolständige oder leere Codes werden nicht gespeichert.
+
+        Returns:
+            Liste mit allen vollständigen Vermittlungscodes
+
+        """
+        codes = []
+        warn = False
+
+        line_edits = self.vermittlungscodes_tab.findChildren(QtWidgets.QLineEdit)
+        for line_edit in line_edits:
+            if len(line_edit.text().strip()) == 14:
+                codes.append(line_edit.text().strip())
+            elif line_edit.text().strip() != "--":
+                warn = True
+
+        if warn:
+            self.__oeffne_error(title="Kontaktdaten", text="Ungültiger Code",
+                                info="Mindestens einer der eingegebenen Codes "
+                                "ist ungültig und wird nicht gespeichert.")
+
+        return codes
```

Comment:
Eigentlich ist eine Überprüfung hier nicht notwendig, da in Zeile 124 die eingelesen Daten alle überprüft werden.

Hier einfach nur leere Codes nicht adden was `"--"` ist. `.trim()` ist nicht nötig, da du bereits mit InputMasks arbeitest

```suggestion
        codes = []

        line_edits = self.vermittlungscodes_tab.findChildren(QtWidgets.QLineEdit)
        for line_edit in line_edits:
            code = line_edit.text()

            # Nur wenn Code nicht leer ist hinzufügen
            if code != "--":
                codes.append(line_edit.text())

        return codes
```

Reply:
Dann arbeitet man mit den bereits vorhandenem und muss nicht doppelt das gleiche überprüfen :)

Reply:
Das sieht gut aus,
Mit den Änderungen ist es aber möglich, eine leere Liste an Codes zu übergeben.
Diese übersteht die Validierung und wird gespeichert. Ist das Verhalten so gewünscht?
Ansonsten würde ich die Validierungsfnktion noch anpassen (Raise exception be Länge 0).

Reply:
Hmm stimmt, wenn man gar keine einträgt ist die Liste leer.
An sich sollte das Valide bleiben im Hinblick auf die Änderungen der Code Gen.

Also wenn du mit "Validierungsfunktion anpassen" die `check_kontaktdaten` bei `kontakdaten.py` meinst, dann stimmt ich dir zu. Hier sollte man nicht nur schauen ob `kontaktdaten["codes"]` keinen `KeyError` wirft, sondern auch zusätzlich  die länge > 0 hat und dann einfach ne `raise KeyError` wirft wenn es nicht der Fall ist

Reply:
Genau das meinte ich, passe ich eben an :)

Reply:
Ich würde das aber als ValidationError implementieren. Der Key ist ja vorhanden. Nur eben nicht gültig.